### PR TITLE
bitmath: init at 1.3.1.2

### PR DESCRIPTION
**Pull request by Twey**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/bitmath/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/bitmath/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/bitmath/default.nix]: Start with a capital letter. The part within parentheses can be dropped.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/bitmath/default.nix]: `checkInputs`

File: pkgs/development/python-modules/bitmath/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/bitmath/default.nix]: should be dropped

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by Twey** [comment]
Thanks!

**Comment by Twey** [comment]
Is this okay?

**Comment by dotlambda** [comment]
No, obviously not:
@GrahamcOfBorg build python3.pkgs.progressbar

**Comment by FRidh** [comment]
Seems like it should be disabled for Python 3.

**Comment by dotlambda** [comment]
But their `setup.py` explicitly mentions Python 3: https://github.com/niltonvolpato/python-progressbar/blob/master/setup.py#L50

**Comment by Twey** [comment]
It might just not support Python 3.6 — the last commit was a couple of years ago.
What's the conventional way to say ‘not on Python 3’?

**Comment by Twey** [comment]
Apparently it's a [known problem](https://github.com/niltonvolpato/python-progressbar/issues/28).  There's a fork with Python 3 support, but it doesn't seem to be on PyPI.  There's also `progressbar2`, which I think is the correct way forward, but it's not quite API-compatible and requires patches to the `bitmath` test suite.

I've just disabled `progressbar` for Python 3 for now.

**Comment by dotlambda** [comment]
Sorry I forgot about this one.
Please remove the progressbar bump as it is included in #38483.
As soon as that PR is merged, you can rebase on staging and this can be merged.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/bitmath/default.nix]: Not necessary because progressbar is now disabled on python3.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/bitmath/default.nix]: Please use `"` and a single line.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/bitmath/default.nix]: no quotes

File: pkgs/development/python-modules/bitmath/default.nix
Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: The line is 94 characters long if I stuff it all onto a single line; is that okay?

File: pkgs/development/python-modules/bitmath/default.nix
Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: I believe it's still necessary, because this package's tests expect progressbar, so if it's disabled on Py3 the tests will fail.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: It's okay for me. I don't think there's a general limit of 80 characters for nixpkgs.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: It won't even be built on Python 3, so no tests will be run. Just rebase on staging and you'll see what I mean.

Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: In fact it doesn't even get to failing the tests — it will try to load `progressbar` for `checkInputs` and immediately fail.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: Fixed.

File: pkgs/development/python-modules/bitmath/default.nix
Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: That's what I meant with "it won't be built". Please just remove that `doCheck = false` line.

Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: But the package itself works just fine with Python 3.6.  Only the test suite fails.

Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: I see. Looking at https://github.com/tbielawa/bitmath/blob/master/requirements.txt and https://github.com/tbielawa/bitmath/blob/master/requirements-py3.txt, it seems like `progressbar231` is required for Python 2 and `progressbar33` for Python 3. At least the latter should be packaged so that tests can be enabled for Python 3.
Btw, please use `isPy3k` instead of `isPy36`.

Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: It's because `progressbar` has been abandoned by its maintainer. Other people created newer versions, which diverged from each other.

Comment (Twey) [reply pkgs/development/python-modules/bitmath/default.nix]: Well, I've packaged them up anyway.

Comment (dotlambda) [reply pkgs/development/python-modules/bitmath/default.nix]: Thanks a lot!

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.progressbar231 python2.pkgs.progressbar33 python3.pkgs.progressbar33 python2.pkgs.bitmath python3.pkgs.bitmath

File: pkgs/development/python-modules/progressbar231/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/progressbar231/default.nix]:
```
# no tests implemented
doCheck = false;
```

File: pkgs/development/python-modules/progressbar33/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/progressbar33/default.nix]:
```
# no tests implemented
doCheck = false;
```

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please add an empty line.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by Twey** [comment]
Done!

**Comment by dotlambda** [comment]
Thanks a lot!

**Pull request merged**
